Mathlib/MeasureTheory/Group/Measure.lean

```diff
@@ -246,6 +247,26 @@ end MeasurableMul
 
 end Mul
 
+section Monoid
+
+variable [Monoid G] [MeasurableMul G]
+
+/-- The image of a left invariant measure under right multiplication is left invariant.
+-/
+@[to_additive isAddLeftInvariant_radd
+"The image of a left invariant measure under right addition is left invariant."]
+theorem isMulLeftInvariant_rmul [IsMulLeftInvariant μ] (g : G) :
+    IsMulLeftInvariant (map (· * g) μ) := by
+  refine' ⟨fun h => _⟩
+  rw [map_map (measurable_const_mul _) (measurable_mul_const _)]
+  conv_rhs => rw [← map_mul_left_eq_self μ h]
+  rw [map_map (measurable_mul_const _) (measurable_const_mul _)]
+  congr 2
+  ext y
+  simp only [comp_apply, mul_assoc h y g]
```

Comment:
This should be possible to prove as a special case of `isMulLeftInvariant_smul` with `MulOpposite`; do we already have that `SMul` version?

Reply:
I am not sure about this? The measure I refer to is `A -> \mu(A g^-1)`, and to my understanding, `smul` is the measure `A -> c \mu(A)`.

Reply:
The statement I'm thinking of is something like
```
theorem SMulInvariantMeasure_smul
    {α β} [MeasurableSpace α] [MeasurableSpace β] [SMul α G] [SMul β G] [SMulCommClass α β G]
    [MeasurableSMul α G] [MeasurableSMul β G] [SMulInvariantMeasure β G μ] (a : α) :
    SMulInvariantMeasure β G (map (a • ·) μ) :=```

Reply:
@urkud is probably more qualified here than me

Reply:
```suggestion
  refine' ⟨fun h => _⟩
  rw [map_map (measurable_const_mul _) (measurable_mul_const _)]
  conv_rhs =>
    rw [← map_mul_left_eq_self μ h, map_map (measurable_mul_const _) (measurable_const_mul _)]
  congr 2 with y
  simp only [comp_apply, mul_assoc h y g]
```

Reply:
```suggestion
  refine' ⟨fun h => _⟩
  rw [map_map (measurable_const_mul _) (measurable_mul_const _)]
  conv_rhs =>
    rw [← map_mul_left_eq_self μ h, map_map (measurable_mul_const _) (measurable_const_mul _)]
  congr 2 with y
  simp only [comp_apply, mul_assoc h y g]
```

Reply:
I see. I think the even more general case is the following. If M, N are G-modules, M equipped with a G-invariant measure `\mu` and `f: M \to N` is a homomorphism, then `f_* \mu` is invariant.

Reply:
I'll write it out and get this as a special case.